src/aggregators/direct.jl

```diff
@@ -58,16 +58,16 @@ end
 # calculate the next jump / jump time
 function generate_jumps!(p::DirectJumpAggregation, integrator, u, params, t)
     p.sum_rate, ttnj = time_to_next_jump(p, u, params, t)
-    @fastmath p.next_jump_time = t + ttnj
+    p.next_jump_time = t + ttnj
     @inbounds p.next_jump = searchsortedfirst(p.cur_rates, rand(p.rng) * p.sum_rate)
     nothing
 end
 
 ######################## SSA specific helper routines ########################
 
 # tuple-based constant jumps
-@fastmath function time_to_next_jump(p::DirectJumpAggregation{T, S, F1, F2, RNG}, u, params,
```

Comment:
In this function we do a cumulative sum and one division at the end (unless the `@fastmath` propagates into the rate and `randexp` functions, in which case it would apply within them too).

Reply:
`@fastmath` doesn't propagate: it's a high level find-replace https://github.com/JuliaLang/julia/blob/master/base/fastmath.jl#L34-L85 so it's not like the C fastmath in that sense. It would need to use something like Cassette to do that anyways.

Reply:
I guess it can set compile flags too? I haven't seen that (https://github.com/JuliaLang/julia/blob/master/base/fastmath.jl#L141-L143), but we can also just directly use add_fast and such.

Reply:
Sure, I can change the relevant floating point additions in these functions to use `add_fast`. We can get `Inf`s for the rates (and hence the cumulative sum), so as long as that propagates correctly we should be fine (and it seemed to propagate correctly before even with `@fastmath`).